src/ubase/trace.ml

```diff
@@ -121,7 +125,16 @@ let logfile =
 
 let logch = ref None
 
+let closelog _ =
+  match !logch with
+    None -> ()
+  | Some(ch,file) ->
+      close_out ch;
+      logch := None;
+  ;;
+
 let rec getLogch() =
+  ignore(signal sigusr1 (Signal_handle closelog));
```

Comment:
Why set the signal handler over and over again instead of setting it once (e.g. like you currently set Signal_ignore)?

Reply:
Also, just so you know, you can use `Sys.set_signal`. Then you don't have to use `ignore` here.

Reply:
That would be fine as well. Given my total lack of experience in OCaml, I just went with this solution, as it compiled and worked after spending a bit too much time on it.

I have revisited the code and changed it to install the signal handler simply once. The commit has been pushed onto this pull request.